### ISSUE TITLE
Training CUDA Out Of Memory error

### ISSUE THREAD

**Issue by Kev1MSL** [TencentARC/InstantMesh#98]
Hi! I am trying to train the instantmesh model but I am currently facing issues just before the backpropagation where I am getting cuda out of memory error. Have you faced a similar issue when training and how did you solve this?
Thanks!

**Comment by gaodalii** [comment]
I am using a single A800(80G),  but I can only train it with batch_size=1, if I set batch_size=2, there also would be a cuda out of memory error.
![image](https://github.com/TencentARC/InstantMesh/assets/158037909/e4ede9ff-71fd-4ca2-8a4b-bd775891bfb7)
![image](https://github.com/TencentARC/InstantMesh/assets/158037909/3ab684c7-d131-4e79-9ed0-8c8495ad0a3b)

**Comment by Kev1MSL** [comment]
Yes same thing, when I set batch_size=1 it works, but batch_size=2 it does not. However I am only missing a few GB (~2GB), so I was wondering if there is a way to optimize this? And also what happens if I want to distribute the training across multiple gpus, if I set batch_size=1, is it going to be 1 batch per GPU? Or the 1 batch will be distributed across the GPUs?

**Comment by Mrguanglei** [comment]
@Kev1MSL  Hello, I encountered several problems in the training process, the structure of my dataset is as the picture says, but my training profile will not be written, I would like to ask for your help, thank you very much for your reply
![Uploading 微信图片_20240531213146.png…]()
![Uploading 微信图片_20240531213352.png…]()

**Comment by throb081** [comment]
@Kev1MSL hello,i am trying to run the training process,but i don't know how to construct the dataset ,can i have a look at the structure of dataset?thank you very much for your reply

**Comment by fffh1** [comment]
> I am using a single A800(80G), but I can only train it with batch_size=1, if I set batch_size=2, there also would be a cuda out of memory error. ![image](https://private-user-images.githubusercontent.com/158037909/335449040-e4ede9ff-71fd-4ca2-8a4b-bd775891bfb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNzc1MzUsIm5iZiI6MTcyMDE3NzIzNSwicGF0aCI6Ii8xNTgwMzc5MDkvMzM1NDQ5MDQwLWU0ZWRlOWZmLTcxZmQtNGNhMi04YTRiLWJkNzc1ODkxYmZiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNVQxMTAwMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MjNmOTY2ZDBlN2Y2M2JiMTBlMWY2YjA3ZWI1ZjExNzVmNWE2MWU0YTgwMThmMmM4OWU0MzFmODY2ZWUxZWZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.WLwA9ZmiM0whXrAeu82khjsouCf--udzrxBDSIjxWfI) ![image](https://private-user-images.githubusercontent.com/158037909/335449230-3ab684c7-d131-4e79-9ed0-8c8495ad0a3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNzc1MzUsIm5iZiI6MTcyMDE3NzIzNSwicGF0aCI6Ii8xNTgwMzc5MDkvMzM1NDQ5MjMwLTNhYjY4NGM3LWQxMzEtNGU3OS05ZWQwLThjODQ5NWFkMGEzYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNVQxMTAwMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OWEzNjE5YWFiNzVmNzU2Y2VlODkyMzY1YzhlMjExNGUxOTJmYTIzZGU2M2QyOWQ5Mzc1M2RiZGVmYjYzYjRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jFZk_5Xn-ZFNwv2roJWb67j4jBpeJCHwT6E0rIW9SOU)

Hello, may I ask anything you changed to code? Because, I am training the model on an A100 GPU, not even able to run with batch size =1 
![image](https://github.com/TencentARC/InstantMesh/assets/87968970/ce41584d-3773-4578-8981-a044bc17d28e)

**Comment by fffh1** [comment]
@Kev1MSL  Hello, may I ask have you made any change to the code ? because I am training the model on the A100 GPU, not even able to train with batch size =1.
![image](https://github.com/TencentARC/InstantMesh/assets/87968970/75e9ac65-8680-4320-acd4-60c46bda5687)

**Comment by ustbzgn** [comment]
@fffh1 Hello, did you solve it? I meet the same problem

**Comment by fffh1** [comment]
Hi
Check your depth image dimension, shuold be one rather than rgb or rgba.
Regards,
Feng
________________________________

**Comment by ustbzgn** [comment]
I am very thanks

---- Replied Message ----
| From | feng ***@***.***> |
| Date | 07/13/2024 18:07 |
| To | TencentARC/InstantMesh ***@***.***> |
| Cc | ustbzgn ***@***.***>,
Comment ***@***.***> |
| Subject | Re: [TencentARC/InstantMesh] Training CUDA Out Of Memory error (Issue #98) |

Hi
Check your depth image dimension, shuold be one rather than rgb or rgba.
Regards,
Feng
________________________________

**Comment by Jinyiyi3** [comment]
> Yes same thing, when I set batch_size=1 it works, but batch_size=2 it does not. However I am only missing a few GB (~2GB), so I was wondering if there is a way to optimize this? And also what happens if I want to distribute the training across multiple gpus, if I set batch_size=1, is it going to be 1 batch per GPU? Or the 1 batch will be distributed across the GPUs?
> 
> Because if it is a batch of size 1, then wouldn't we have issue with converging?

Hello, may I ask how you figure this 'cuda out of memory' issue? I meet the same question while validating stage, but seems confused to solve it.
